pandas/tests/indexes/multi/test_sorting.py

```diff
@@ -271,3 +271,24 @@ def test_argsort(idx):
     result = idx.argsort()
     expected = idx.values.argsort()
     tm.assert_numpy_array_equal(result, expected)
+
+
+def test_not_remove_nan():
+    # GH 37510
+    df1 = DataFrame(
+        {
+            "id1": [1, 2, 3, 4],
+            "id2": [3, 4, 1, 2],
+            "id3": [1, 1, 1, 1],
+            "x": [1, 2, 3, 4],
+        }
+    )
+    df1 = df1.set_index(["id1", "id2", "id3"])
+    new_levels = ["n1", "n2", "n3", None]
+    df1.index = df1.index.set_levels(levels=new_levels, level="id1")
+    df1.index = df1.index.set_levels(levels=new_levels, level="id2")
+
+    result = df1.unstack("id3")[("x", 1)].sort_index().index
```

Comment:
I'd say construct the expected frame and compare using `assert_frame_equal`

alternatively use `tm.assert_index_equal`